pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/ValidatorTest.java

```diff
@@ -670,6 +670,14 @@ public void parallelStagesDeepNesting() throws Exception {
                 .go();
     }
 
+    @Issue("JENKINS-46809")
+    @Test
+    public void parallelStagesGroupsDeepNesting() throws Exception {
+        expectError("parallelStagesGroupsDeepNesting")
+                .logContains(Messages.ModelValidatorImpl_NoNestedWithinNestedStages())
+                .go();
+    }
+
```

Comment:
I'd love to see a similar test, which checks the ability to override/inherit `agent`, `tools`, `environment`, `when`, and `post`

Reply:
See above in the **TEST TODO**s =)

Reply:
Oh, right. That. :)